itest/lnd_remote_signer_test.go

```diff
@@ -261,7 +266,136 @@ func testRemoteSigner(ht *lntest.HarnessTest) {
+		*node.HarnessNode, *node.HarnessNode) {
+
+		// Signer is our signing node and has the wallet with the full
+		// master private key. We test that we can create the watch-only
+		// wallet from the exported accounts but also from a static key
+		// to make sure the derivation of the account public keys is
+		// correct in both cases.
+		password := []byte("itestpassword")
+		var (
+			signerNodePubKey  = nodePubKey
+			watchOnlyAccounts = deriveCustomScopeAccounts(ht.T)
+			signer            *node.HarnessNode
+			err               error
+		)
+
+		var commitArgs []string
+		if subTest.commitType == lnrpc.CommitmentType_SIMPLE_TAPROOT {
+			commitArgs = lntest.NodeArgsForCommitType(
+				subTest.commitType,
+			)
+		}
+
+		// WatchOnly is the node that has a watch-only wallet and uses
+		// the Signer node for any operation that requires access to
+		// private keys. We use the outbound signer type here, meaning
+		// that the watch-only node expects the signer to make an
+		// outbound connection to it.
+		watchOnly := st.CreateNewNode(
+			"WatchOnly", append([]string{
+				"--remotesigner.enable",
+				"--remotesigner.signertype=outbound",
+				"--remotesigner.timeout=30s",
+				"--remotesigner.requesttimeout=30s",
+			}, commitArgs...),
+			password,
+		)
+
+		// As the signer node will make an outbound connection to the
+		// watch-only node, we must specify the watch-only node's RPC
+		// connection details in the signer's configuration.
+		signerArgs := []string{
+			"--remotesigner.signertype=signer", // Outbound signer.
+			"--remotesigner.timeout=30s",
+			"--remotesigner.requesttimeout=10s",
+			fmt.Sprintf(
+				"--remotesigner.rpchost=localhost:%d",
+				watchOnly.Cfg.RPCPort,
+			),
+			fmt.Sprintf(
+				"--remotesigner.tlscertpath=%s",
+				watchOnly.Cfg.TLSCertPath,
+			),
+			fmt.Sprintf(
+				"--remotesigner.macaroonpath=%s",
+				watchOnly.Cfg.AdminMacPath,
+			),
+		}
+
+		if !subTest.randomSeed {
+			signer = st.RestoreNodeWithSeed(
+				"Signer", signerArgs, password, nil, rootKey, 0,
+				nil,
+			)
+		} else {
+			signer = st.NewNode("Signer", signerArgs)
+			signerNodePubKey = signer.PubKeyStr
+
+			rpcAccts := signer.RPC.ListAccounts(
+				&walletrpc.ListAccountsRequest{},
+			)
+
+			watchOnlyAccounts, err = walletrpc.AccountsToWatchOnly(
+				rpcAccts.Accounts,
+			)
+			require.NoError(st, err)
+		}
+
+		// As the watch-only node will not fully start until the signer
+		// node connects to it, we need to start the watch-only node
+		// after having started the signer node.
+		st.StartWatchOnly(watchOnly, "WatchOnly", password,
+			&lnrpc.WatchOnly{
+				MasterKeyBirthdayTimestamp: 0,
+				MasterKeyFingerprint:       nil,
+				Accounts:                   watchOnlyAccounts,
+			},
+		)
+
+		resp := watchOnly.RPC.GetInfo()
+		require.Equal(st, signerNodePubKey, resp.IdentityPubkey)
+
+		if subTest.sendCoins {
+			st.FundCoins(btcutil.SatoshiPerBitcoin, watchOnly)
+			ht.AssertWalletAccountBalance(
```

Comment:
I notice there's seemingly no restart required here. Is that because we programatically pass along all the relevant connection details?